funnel/models/space.py

```diff
@@ -109,24 +120,26 @@ def proposals_all(self):
             return self.proposals
 
     @property
-    def proposals_by_status(self):
-        from .proposal import Proposal, PROPOSALSTATUS
+    def proposals_by_state(self):
+        from .proposal import Proposal
         if self.subspaces:
             basequery = Proposal.query.filter(Proposal.proposal_space_id.in_([self.id] + [s.id for s in self.subspaces]))
         else:
             basequery = Proposal.query.filter_by(proposal_space=self)
-        return dict((status, basequery.filter_by(status=status).order_by(db.desc('created_at')).all()) for (status, title) in PROPOSALSTATUS.items() if status != PROPOSALSTATUS.DRAFT)
+        all_proposals = basequery.filter(~Proposal.state.DRAFT).order_by(db.desc('created_at'))
+        all_by_state = Proposal.state.group(all_proposals)
```

Comment:
There's perhaps no need for `all_proposals` and `all_by_state` here, you can return the result of the query directly.

Reply:
I think I like assigning variables for more readability. Is there any harm of using them?

Reply:
Assigning variables is _slightly_ slower because it has to allocate memory for that variable in the local namespace, and then reference it again in the next statement. There is also the potential for developer error when a name accidentally clobbers another variable with that name, or if there are two similarly named variables and the developer accidentally refers to the wrong one. I find myself making this error all the time (usually caught by failing tests).

As a matter of habit, I avoid giving names and instead leave a comment.

That said, names are useful when debugging because it helps examine each line separately, so I do this during development and remove the names later.

Reply:
Got it. Cool. I'll update the code.